python/cudf/cudf/core/dataframe.py

```diff
@@ -3555,7 +3555,7 @@ def isin(self, values):
             for col in self.columns:
                 if col in values:
                     val = values[col]
-                    result_df[col] = self[col].isin(val)
+                    result_df[col] = self[col].isin(val)._column
```

Comment:
This just reaches underneath the result Series after doing the `isin` and Index handling within `isin`. We should be able to do something like `self._data[col].isin(val)` for example where the `isin` is run against the Column as opposed to being run against the Series (and subsequently performing unnecessary index handling).

Reply:
Added `Column.isin` (without index handling) and handling only in `DataFrame` & `Series` calls individually.